### DISCUSSION TITLE
SIGTERM / Scaling down - pod still working and not have new traffic

### DISCUSSION THREAD

**Discussion by sr185436** [kedacore/keda#5280] [Q&A / Need Help]
Hello,

When "Keda" Scaling down how i can control & avoid of scaling down pods that still processing... for example situation that "pod" still working and processing (not have new traffic/request ...)?
cooldownPeriod or Grace value... not solve issues like that..
if pod still work and not have new traffic, what is the recommend configuration of keda?

Thanks !

**Comment by JorTurFer** [comment]
Hello @sr185436 
You can't control that as it's not KEDA who decides the pods to be removed, it's the HPA controller who does it. If you need more control about the removed pods. If your scenario an HTTP scenario or it's related with in-progress message processing?

**Reply by sr185436** [reply]
Our concern is about situation that "pod" still working and processing (we don't have new traffic/requests but the pod still work/processing ...).

cooldownPeriod or Grace value, not solve issues like that.. 

Hope that you can help us

**Reply by JorTurFer** [reply]
> You can't control that as it's not KEDA who decides the pods to be removed, it's the HPA controller who does it.

There is nothing KEDA can do to decide which pods have to be removed by the HPA Controller but understanding what you understand as new traffic is important for helping you with a solution to mitigate this.
When you say `new traffic/requests`, do you mean HTTP requests or consuming messages? In case of HTTP traffic, you have to handle it increasing the graceful termination period to bring enough time to finish the current requests. 
In case of messages is more complicated and it depends on 2 factors:
- How much time is needed for processing the message?
- Has the message ACK?

With your answers, we can try to go forward
